bitmind/image_transforms.py

```diff
@@ -9,6 +10,80 @@ def fn(img):
     return fn
 
 
+class RandomResizedCropWithParams(transforms.RandomResizedCrop):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.params = None
+
+    def get_params(self, img, scale, ratio):
+        params = super().get_params(img, scale, ratio)
+        self.params = params
+        return params
+
+
+class RandomHorizontalFlipWithParams(transforms.RandomHorizontalFlip):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.params = None
+
+    def forward(self, img):
+        if torch.rand(1) < self.p:
+            self.params = True
+            return transforms.functional.hflip(img)
+        else:
+            self.params = False
+            return img
+
+
+class RandomVerticalFlipWithParams(transforms.RandomVerticalFlip):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.params = None
+
+    def forward(self, img):
+        if torch.rand(1) < self.p:
+            self.params = True
+            return transforms.functional.vflip(img)
+        else:
+            self.params = False
+            return img
+
+
+class RandomRotationWithParams(transforms.RandomRotation):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.params = None
+
+    def forward(self, img):
+        angle = self.get_params(self.degrees)
+        self.params = angle
+        return transforms.functional.rotate(img, angle)
+
+
+class ComposeWithParams:
```

Comment:
Looks good! Small cosmetic suggestion for ComposeWithParams

def __call__(self, img):
    transform_params = {
        RandomResizedCropWithParams: 'RandomResizedCrop',
        RandomHorizontalFlipWithParams: 'RandomHorizontalFlip',
        RandomVerticalFlipWithParams: 'RandomVerticalFlip',
        RandomRotationWithParams: 'RandomRotation'
    }
    
    for t in self.transforms:
        img = t(img)
        if type(t) in transform_params:
            self.params[transform_params[type(t)]] = t.params
    
    return img

Reply:
great suggestion, updating now